src/operator/nn/mkldnn/mkldnn_reshape.cc

```diff
@@ -33,6 +33,13 @@
 namespace mxnet {
 namespace op {
 
+bool SupportMKLDNNReshape(const NDArray& input, const NDArray& output) {
+  const int input_ndims  = input.shape().ndim();
+  const int output_ndims = output.shape().ndim();
+  return input_ndims >= 1 && input_ndims <= 6 && output_ndims >= 1 && output_ndims <= 6 &&
+         IsMKLDNNType(input.dtype()) && input.shape().Size() > 0;
```

Comment:
The input is having its size checked at the end of this statement ~ if the Size() function returns the size of the input, then it could be better to place it at the beginning.